File: libs/csharp/common/src/Core/Exceptions/NotSupportedParsingException.cs
namespace Crosslight.Core.Exceptions;

/// <summary>
/// This node configuration is not and will not be supported.
/// </summary>
public class NotSupportedParsingException : ParsingException
{
    public NotSupportedParsingException(Node errorNode)
        : base(errorNode)
    {
    }

    public NotSupportedParsingException(string message, Node errorNode)
        : base(message, errorNode)
    {
    }

    public NotSupportedParsingException(string message, Exception inner, Node errorNode)
        : base(message, inner, errorNode)
    {
    }
}

File: libs/cpp/lang/src/exceptions/not_supported_parsing_exception.cpp
#include "lang/exceptions/not_supported_parsing_exception.hpp"

using namespace cl::lang::exceptions;

not_supported_parsing_exception::not_supported_parsing_exception(const std::string& what_arg)
    : parsing_exception{what_arg} {}

not_supported_parsing_exception::not_supported_parsing_exception(const char* what_arg)
    : parsing_exception{what_arg} {}

const char*
not_supported_parsing_exception::what() const noexcept {
    return _message.c_str();
}
